util/src/main/java/io/grpc/util/LruCache.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.util;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+public class LruCache<K, V> {
+
+  private final Map<K, V> cache;
+  private final long maxSize;
+
+  public LruCache(long maxSize) {
+    this.maxSize = maxSize;
+    this.cache = new LinkedHashMap<K, V>(
+        (int) Math.min(maxSize, Integer.MAX_VALUE),
+        0.75f,
+        true) {
+      @Override
+      protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
+        return size() > maxSize;
```

Comment:
FYI, this is fine for now, but later you're going to need to change the maxSize after construction.